internal/command/jsonformat/renderer.go

```diff
@@ -240,6 +264,49 @@ func (r Renderer) RenderHumanPlan(plan Plan, mode plans.Mode, opts ...RendererOp
 	}
 }
 
+func (r Renderer) RenderLog(log *JSONLog) error {
+	switch log.Type {
+	case LogApplyStart, LogApplyComplete, LogRefreshStart:
+		msg := fmt.Sprintf("[bold]%s[reset]", log.Message)
+		r.Streams.Println(r.Colorize.Color(msg))
+
+	case LogDiagnostic:
+		diag := format.DiagnosticFromJSON(log.Diagnostic, r.Colorize, 78)
+		r.Streams.Print(diag)
+
+	case LogOutputs:
+		if len(log.Outputs) > 0 {
+			r.Streams.Println(r.Colorize.Color("[bold][green]Outputs:[reset]"))
+			for name, output := range log.Outputs {
+				change := differ.FromJsonOutput(output)
+				ctype, err := ctyjson.UnmarshalType(output.Type)
+				if err != nil {
+					return err
+				}
+
+				outputDiff := change.ComputeDiffForType(ctype)
+				outputStr := outputDiff.RenderHuman(0, computed.RenderHumanOpts{
+					Colorize:              r.Colorize,
+					ShowUnchangedChildren: true,
+				})
+
+				msg := fmt.Sprintf("%s = %s", name, outputStr)
+				r.Streams.Println(r.Colorize.Color(msg))
+			}
+		}
+
+	case LogChangeSummary:
+		// We will only render the apply change summary since the renderer
+		// generates a plan change summary for us
+		if !strings.Contains(log.Message, "Plan") {
```

Comment:
This feels like a hack. Is this necessary, or can we prevent calling `RenderLog` with this type when we're in plan mode?